src/thirdparty/ALIKED/get_patches.cpp

```diff
@@ -0,0 +1,150 @@
+#include "get_patches_cuda.h"
+#include <glog/logging.h>
+#include <math.h>
+#include <torch/torch.h>
+
+// map: CxHxW
+// points: Nx2
+// kernel_size: int
+// return: N x C x kernel_size x kernel_size
+namespace custom_ops {
+torch::Tensor get_patches_forward_cpu(const torch::Tensor& map,
+                                      torch::Tensor& points,
+                                      int64_t kernel_size) {
+  namespace F = torch::nn::functional;
+  using namespace torch::indexing;
+
+  auto N = points.size(0);
+  auto C = map.size(0);
+  // kernel_size=2, radius=0.5, pad_left_top=0, pad_right_bottom=1
+  // kernel_size=3, radius=1.0, pad_left_top=1, pad_right_bottom=1
+  // kernel_size=4, radius=1.5, pad_left_top=1, pad_right_bottom=2
+  // kernel_size=5, radius=2.0, pad_left_top=2, pad_right_bottom=2
+  auto radius = (kernel_size - 1.0) / 2.0;
+  int pad_left_top = floor(radius);
+  int pad_right_bottom = ceil(radius);
+
+  // pad map
+  auto options =
+      F::PadFuncOptions(
+          {pad_left_top, pad_right_bottom, pad_left_top, pad_right_bottom})
+          .mode(torch::kConstant);
+  auto map_pad = F::pad(map.unsqueeze(0), options)
+                     .squeeze(0);  // Cx(H+2*radius)x(W+2*radius)
+
+  // get patches
+  torch::Tensor patches =
+      torch::zeros({N, C, kernel_size, kernel_size}, map.options());
+  auto a_points = points.accessor<int, 2>();     // Nx2
```

Comment:
In a bunch of places, your original code here had `.accessor<long, 2>()` and I had to change it to `int` to avoid missing symbol errors.

Reply:
Interesting. I didn't check the code as it is taken from the original implementation https://github.com/Shiaoming/ALIKED/blob/main/custom_ops/get_patches.cpp#L33.

```
 torch::Tensor get_patches_forward(const torch::Tensor& map, torch::Tensor& points, int64_t kernel_size) {
        if (map.device() == torch::kCPU)
            return get_patches_forward_cpu(map, points, kernel_size);
        else
        {
            return get_patches_forward_cuda(map, points, kernel_size);
        }
    }
```
    
So, the points should have the same data type as in the cuda path. The data is accessed via points.data_ptr<int64_t>() in the cuda version. I guess accessing it as pointer to int would be a severe bug? 

I am not 100% sure why there are missing symbols? Does int64_t work?

Reply:
Ok, while thinking more about it, I am confused. Accessor is a safe way to access the tensor data, data_ptr is not. So if there are compile time issues with the datatype, something is really wrong. Maybe the cuda part is buggy but I didn't encounter with cuda any issues. Ok, I think I have too take a closer look into it.

Reply:
I guess you are using the libtorch cpu verision. I only used exclusively the cuda 2.1 version which is why never ran into these issues.